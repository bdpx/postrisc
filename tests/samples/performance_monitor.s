.text
    alloc	21
    ldi		%r12, PMC_LAST
    write	"PMC_LAST = %i64(r12)"
; don't report runtine in unittests, this is non-reproducible
    mf.mr	%r14, %gz, PMC_RUNTIME
;   write	"PMC_RUNTIME = %i64(r14)"
    mf.mr	%r14, %gz, PMC_SHORT_INSTRUCTION
    write	"PMC_SHORT_INSTRUCTION = %i64(r14)"
    mf.mr	%r14, %gz, PMC_LONG_INSTRUCTION
    write	"PMC_LONG_INSTRUCTION = %i64(r14)"
    mf.mr	%r14, %gz, PMC_SHADOWED_INSTRUCTION
    write	"PMC_SHADOWED_INSTRUCTION = %i64(r14)"
    mf.mr	%r14, %gz, PMC_NOP_INSTRUCTION
    write	"PMC_NOP_INSTRUCTION = %i64(r14)"
    mf.mr	%r14, %gz, PMC_QUALIFIED_NOP_INSTRUCTION
    write	"PMC_QUALIFIED_NOP_INSTRUCTION = %i64(r14)"
    mf.mr	%r14, %gz, PMC_REGISTER_SPILL
    write	"PMC_REGISTER_SPILL = %i64(r14)"
    mf.mr	%r14, %gz, PMC_REGISTER_FILL
    write	"PMC_REGISTER_FILL = %i64(r14)"
    mf.mr	%r14, %gz, PMC_ICACHE_HIT
    write	"PMC_ICACHE_HIT = %i64(r14)"
    mf.mr	%r14, %gz, PMC_ICACHE_MISS
    write	"PMC_ICACHE_MISS = %i64(r14)"
    mf.mr	%r14, %gz, PMC_DCACHE_HIT
    write	"PMC_DCACHE_HIT = %i64(r14)"
    mf.mr	%r14, %gz, PMC_DCACHE_MISS
    write	"PMC_DCACHE_MISS = %i64(r14)"
    mf.mr	%r14, %gz, PMC_INSTRUCTION_TRANSLATION_HIT
    write	"PMC_INSTRUCTION_TRANSLATION_HIT = %i64(r14)"
    mf.mr	%r14, %gz, PMC_INSTRUCTION_TRANSLATION_MISS
    write	"PMC_INSTRUCTION_TRANSLATION_MISS = %i64(r14)"
    mf.mr	%r14, %gz, PMC_DATA_TRANSLATION_HIT
    write	"PMC_DATA_TRANSLATION_HIT = %i64(r14)"
    mf.mr	%r14, %gz, PMC_DATA_TRANSLATION_MISS
    write	"PMC_DATA_TRANSLATION_MISS = %i64(r14)"
    mf.mr	%r14, %gz, PMC_BACKSTORE_TRANSLATION_HIT
    write	"PMC_BACKSTORE_TRANSLATION_HIT = %i64(r14)"
    mf.mr	%r14, %gz, PMC_BACKSTORE_TRANSLATION_MISS
    write	"PMC_BACKSTORE_TRANSLATION_MISS = %i64(r14)"
    mt.mr	%r14, %gz, PMC_SHORT_INSTRUCTION
    mf.mr	%r15, %gz, PMC_SHORT_INSTRUCTION
    write	"old pm reg = %i64(r15)"
.end
