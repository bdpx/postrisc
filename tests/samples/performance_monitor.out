PMC_LAST = 20
PMC_SHORT_INSTRUCTION = 5
PMC_LONG_INSTRUCTION = 0
PMC_SHADOWED_INSTRUCTION = 0
PMC_NOP_INSTRUCTION = 0
PMC_QUALIFIED_NOP_INSTRUCTION = 0
PMC_REGISTER_SPILL = 0
PMC_REGISTER_FILL = 0
PMC_ICACHE_HIT = 0
PMC_ICACHE_MISS = 1
PMC_DCACHE_HIT = 0
PMC_DCACHE_MISS = 0
PMC_INSTRUCTION_TRANSLATION_HIT = 0
PMC_INSTRUCTION_TRANSLATION_MISS = 1
PMC_DATA_TRANSLATION_HIT = 458
PMC_DATA_TRANSLATION_MISS = 3
PMC_BACKSTORE_TRANSLATION_HIT = 0
PMC_BACKSTORE_TRANSLATION_MISS = 0
old pm reg = 1
